File: 0110-balanced-binary-tree/0110-balanced-binary-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {boolean}
 */
var isBalanced = function(root) {
  
    const checkHeight = (node) =>{
        if(node == null) return 0;
        
        let leftH = checkHeight(node.left);
        if(leftH == -1) return -1;
        
        let rightH = checkHeight(node.right);
        if(rightH == -1) return -1;
        
        if(Math.abs(rightH - leftH) > 1){
            return -1
        }
        
        return Math.max(leftH, rightH)+1;
    }
    
    return checkHeight(root) != -1;
};

File: 0110-balanced-binary-tree/0110-balanced-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public boolean isBalanced(TreeNode root) {
        return checkHeight(root) != -1;
    }
    
    private int checkHeight(TreeNode node){
        if(node == null) return 0;
        
        int leftH = checkHeight(node.left);
        if(leftH == -1) return -1;
            
        int rightH = checkHeight(node.right);
        if(rightH == -1) return -1;
        
        if(Math.abs(leftH - rightH) > 1){
            return -1;
        }
        
        return Math.max(leftH, rightH) + 1;
    }
}